In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
df=pd.read_csv('train.csv')

In [3]:
df.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.186257  -0.058807   0.024632  -0.163933  -0.146699   0.035889   
1   0.063431  -0.023597   0.068057  -0.252915  -0.061094  -0.027316   
2   0.034962   0.035816  -0.029753  -0.094607  -0.017576  -0.053074   
3   0.033772   0.085612   0.067488  -0.073953  -0.180646  -0.024512   
4   0.134305   0.062096   0.106920  -0.089327   0.117093  -0.077107   

   feature_7  feature_8  feature_9  feature_10  ...  feature_763  feature_764  \
0   0.111708  -0.162861   0.028249   -0.098063  ...     0.055629    -0.010358   
1   0.135747  -0.168147   0.091236   -0.078473  ...    -0.014893     0.071721   
2   0.040121  -0.007932   0.097872   -0.024042  ...     0.012415     0.015215   
3   0.242879  -0.023374  -0.059999    0.002006  ...    -0.078246    -0.032903   
4   0.152579   0.047529  -0.015998   -0.110657  ...    -0.094629     0.069718   

   feature_765  feature_766  feature_767  feature_768  label_1  label_2  \
0     0.125754     0.011648     0.079197     0.093215       45      NaN   
1     0.018918     0.100032    -0.083042     0.088615       45      NaN   
2     0.083808     0.031312    -0.056277     0.064702       45      NaN   
3     0.082949    -0.020659     0.082274    -0.050164       45      NaN   
4     0.014379     0.048124     0.007586    -0.016980       45      NaN   

   label_3  label_4  
0        1        6  
1        1        6  
2        1        6  
3        1        6  
4        1        6  

[5 rows x 772 columns]

In [4]:
##check null values
df.isnull().sum()

feature_1        0
feature_2        0
feature_3        0
feature_4        0
feature_5        0
              ... 
feature_768      0
label_1          0
label_2        480
label_3          0
label_4          0
Length: 772, dtype: int64

In [5]:
y_train = df[["label_1","label_2","label_3","label_4"]]
X_train = df.drop(y_train, axis=1)


In [6]:
X_train

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0       0.186257  -0.058807   0.024632  -0.163933  -0.146699   0.035889   
1       0.063431  -0.023597   0.068057  -0.252915  -0.061094  -0.027316   
2       0.034962   0.035816  -0.029753  -0.094607  -0.017576  -0.053074   
3       0.033772   0.085612   0.067488  -0.073953  -0.180646  -0.024512   
4       0.134305   0.062096   0.106920  -0.089327   0.117093  -0.077107   
...          ...        ...        ...        ...        ...        ...   
28515   0.024453   0.044466  -0.049367  -0.065155  -0.109931  -0.089393   
28516   0.016492   0.007850   0.004661  -0.068151  -0.120102  -0.018188   
28517   0.024608   0.045056   0.011118  -0.055420  -0.180276  -0.075961   
28518   0.174154  -0.068870   0.058101  -0.120567  -0.237380   0.024585   
28519   0.069458   0.102667  -0.010687  -0.054117  -0.171478  -0.154001   

       feature_7  feature_8  feature_9  feature_10  ...  feature_759  \
0       0.111708  -0.162861   0.028249   -0.098063  ...    -0.135802   
1       0.135747  -0.168147   0.091236   -0.078473  ...    -0.032524   
2       0.040121  -0.007932   0.097872   -0.024042  ...    -0.016680   
3       0.242879  -0.023374  -0.059999    0.002006  ...    -0.219269   
4       0.152579   0.047529  -0.015998   -0.110657  ...    -0.228570   
...          ...        ...        ...         ...  ...          ...   
28515   0.003823   0.036797   0.142065    0.010630  ...     0.024859   
28516   0.088983  -0.016272   0.057924   -0.091656  ...    -0.067817   
28517   0.047853   0.043180   0.152698   -0.111217  ...    -0.066240   
28518   0.187333  -0.100873   0.039363   -0.080888  ...     0.062747   
28519   0.076675   0.025081   0.118611   -0.061732  ...    -0.130062   

       feature_760  feature_761  feature_762  feature_763  feature_764  \
0         0.101059     0.128649    -0.187129     0.055629    -0.010358   
1         0.012977     0.090799    -0.254056    -0.014893     0.071721   
2        -0.015495    -0.003885    -0.106426     0.012415     0.015215   
3         0.065505     0.016679     0.047519    -0.078246    -0.032903   
4         0.064493     0.177749    -0.117646    -0.094629     0.069718   
...            ...          ...          ...          ...          ...   
28515     0.034621     0.051483     0.055168    -0.037605    -0.009495   
28516    -0.003412     0.120643    -0.156379    -0.043655    -0.065877   
28517     0.146486     0.103236     0.060171     0.045422    -0.109505   
28518     0.092733     0.099706     0.045776     0.028609    -0.081717   
28519     0.153767     0.124336    -0.049092     0.118842    -0.177180   

       feature_765  feature_766  feature_767  feature_768  
0         0.125754     0.011648     0.079197     0.093215  
1         0.018918     0.100032    -0.083042     0.088615  
2         0.083808     0.031312    -0.056277     0.064702  
3         0.082949    -0.020659     0.082274    -0.050164  
4         0.014379     0.048124     0.007586    -0.016980  
...            ...          ...          ...          ...  
28515     0.069628    -0.064233    -0.082123     0.068675  
28516     0.114468     0.085360    -0.032404     0.076920  
28517    -0.000356     0.022698     0.065250     0.017646  
28518     0.089243     0.017368     0.044567     0.089306  
28519     0.004411     0.034947     0.039597     0.031190  

[28520 rows x 768 columns]

In [7]:
y_train

label_1  label_2  label_3  label_4
0           45      NaN        1        6
1           45      NaN        1        6
2           45      NaN        1        6
3           45      NaN        1        6
4           45      NaN        1        6
...        ...      ...      ...      ...
28515       39     29.0        1        6
28516       39     29.0        1        6
28517       39     29.0        1        6
28518       39     29.0        1        6
28519       39     29.0        1        6

[28520 rows x 4 columns]

In [8]:
df.corr()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
feature_1     1.000000  -0.283908   0.142182  -0.338103  -0.178025  -0.207995   
feature_2    -0.283908   1.000000  -0.085842   0.155011   0.011385   0.082116   
feature_3     0.142182  -0.085842   1.000000   0.145121  -0.003237  -0.056097   
feature_4    -0.338103   0.155011   0.145121   1.000000   0.058212  -0.130674   
feature_5    -0.178025   0.011385  -0.003237   0.058212   1.000000   0.060386   
...                ...        ...        ...        ...        ...        ...   
feature_768  -0.030147   0.022819  -0.055686  -0.316624   0.181856   0.009531   
label_1      -0.055811   0.024157   0.098467   0.046853  -0.070505  -0.021496   
label_2      -0.031686   0.037137   0.014603   0.006841   0.103229   0.048361   
label_3       0.041166  -0.035141   0.030263  -0.059081   0.106992   0.041350   
label_4       0.023219   0.025806  -0.051158  -0.008024   0.006422   0.016311   

             feature_7  feature_8  feature_9  feature_10  ...  feature_763  \
feature_1     0.148884   0.052742  -0.315786   -0.155347  ...    -0.002735   
feature_2    -0.069860  -0.000870   0.250169    0.192959  ...     0.257924   
feature_3     0.137698   0.120606  -0.215018   -0.255378  ...    -0.128951   
feature_4    -0.099220   0.064827   0.098506   -0.067288  ...    -0.211635   
feature_5    -0.048629  -0.092161   0.024778    0.219142  ...    -0.210182   
...                ...        ...        ...         ...  ...          ...   
feature_768  -0.398800  -0.131120   0.301206    0.234216  ...     0.221313   
label_1      -0.024910   0.043663  -0.033741   -0.046304  ...    -0.010398   
label_2      -0.023677  -0.006038  -0.000270   -0.015349  ...    -0.009239   
label_3       0.038326   0.091699   0.012572    0.009689  ...     0.052200   
label_4      -0.067295  -0.146392   0.012093    0.026688  ...     0.119500   

             feature_764  feature_765  feature_766  feature_767  feature_768  \
feature_1       0.202713    -0.255096    -0.186352     0.169145    -0.030147   
feature_2      -0.023016    -0.014792    -0.112536    -0.191086     0.022819   
feature_3      -0.317668    -0.209626     0.081598     0.240037    -0.055686   
feature_4      -0.362384    -0.144449    -0.000296    -0.088443    -0.316624   
feature_5       0.148122     0.199165     0.198393    -0.142673     0.181856   
...                  ...          ...          ...          ...          ...   
feature_768     0.142108     0.234996     0.158833    -0.178095     1.000000   
label_1        -0.126097    -0.070339    -0.002718     0.104106    -0.048178   
label_2         0.013713    -0.037036     0.019229    -0.024101     0.018263   
label_3         0.089202     0.137891    -0.022264    -0.045734     0.139805   
label_4         0.049514    -0.015315     0.017489    -0.018674     0.024310   

              label_1   label_2   label_3   label_4  
feature_1   -0.055811 -0.031686  0.041166  0.023219  
feature_2    0.024157  0.037137 -0.035141  0.025806  
feature_3    0.098467  0.014603  0.030263 -0.051158  
feature_4    0.046853  0.006841 -0.059081 -0.008024  
feature_5   -0.070505  0.103229  0.106992  0.006422  
...               ...       ...       ...       ...  
feature_768 -0.048178  0.018263  0.139805  0.024310  
label_1      1.000000 -0.038758 -0.403477 -0.021115  
label_2     -0.038758  1.000000  0.086053  0.242613  
label_3     -0.403477  0.086053  1.000000  0.016825  
label_4     -0.021115  0.242613  0.016825  1.000000  

[772 rows x 772 columns]

In [58]:
valid_df = pd.read_csv('valid.csv')

In [59]:
valid_df.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.085129   0.000378  -0.029872  -0.104609  -0.191291  -0.025511   
1   0.222652  -0.094210  -0.056387  -0.210214  -0.294624  -0.032355   
2   0.072147   0.023575  -0.017664  -0.070988  -0.098729   0.018534   
3   0.026623   0.052005  -0.006921  -0.075105  -0.074829  -0.031380   
4   0.112069   0.113702   0.044343   0.010162  -0.169193  -0.099847   

   feature_7  feature_8  feature_9  feature_10  ...  feature_763  feature_764  \
0   0.029381   0.006177   0.079207   -0.012729  ...    -0.026701     0.003306   
1  -0.069078  -0.172146   0.056509   -0.126757  ...    -0.041096     0.013654   
2   0.028282  -0.064789   0.094686   -0.067313  ...    -0.064602    -0.018653   
3   0.012402  -0.043555   0.073825   -0.050979  ...    -0.005655     0.015477   
4   0.195957   0.051861   0.024582    0.001314  ...    -0.118097    -0.043349   

   feature_765  feature_766  feature_767  feature_768  label_1  label_2  \
0     0.052219     0.000784     0.004527     0.102033       45      NaN   
1     0.142623    -0.011882     0.005219     0.143616       45      NaN   
2     0.013636    -0.078994     0.060903     0.064752       45      NaN   
3     0.064209     0.025045    -0.049599     0.078412       45      NaN   
4     0.036754    -0.023065     0.007358    -0.132566       45      NaN   

   label_3  label_4  
0        1        6  
1        1        6  
2        1        6  
3        1        6  
4        1        6  

[5 rows x 772 columns]

In [11]:
valid_df.isnull().sum()

feature_1       0
feature_2       0
feature_3       0
feature_4       0
feature_5       0
               ..
feature_768     0
label_1         0
label_2        14
label_3         0
label_4         0
Length: 772, dtype: int64

In [12]:
y_valid = valid_df[["label_1","label_2","label_3","label_4"]]
X_valid = valid_df.drop(y_valid, axis=1)

In [13]:
X_valid.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.085129   0.000378  -0.029872  -0.104609  -0.191291  -0.025511   
1   0.222652  -0.094210  -0.056387  -0.210214  -0.294624  -0.032355   
2   0.072147   0.023575  -0.017664  -0.070988  -0.098729   0.018534   
3   0.026623   0.052005  -0.006921  -0.075105  -0.074829  -0.031380   
4   0.112069   0.113702   0.044343   0.010162  -0.169193  -0.099847   

   feature_7  feature_8  feature_9  feature_10  ...  feature_759  feature_760  \
0   0.029381   0.006177   0.079207   -0.012729  ...    -0.020899     0.087778   
1  -0.069078  -0.172146   0.056509   -0.126757  ...    -0.159676     0.134823   
2   0.028282  -0.064789   0.094686   -0.067313  ...    -0.072512     0.082791   
3   0.012402  -0.043555   0.073825   -0.050979  ...     0.012509     0.010695   
4   0.195957   0.051861   0.024582    0.001314  ...    -0.173597     0.247099   

   feature_761  feature_762  feature_763  feature_764  feature_765  \
0     0.083750     0.111214    -0.026701     0.003306     0.052219   
1     0.139934     0.035135    -0.041096     0.013654     0.142623   
2     0.017780     0.030908    -0.064602    -0.018653     0.013636   
3     0.019255     0.010389    -0.005655     0.015477     0.064209   
4     0.101261     0.127192    -0.118097    -0.043349     0.036754   

   feature_766  feature_767  feature_768  
0     0.000784     0.004527     0.102033  
1    -0.011882     0.005219     0.143616  
2    -0.078994     0.060903     0.064752  
3     0.025045    -0.049599     0.078412  
4    -0.023065     0.007358    -0.132566  

[5 rows x 768 columns]

In [14]:
y_valid.head()

label_1  label_2  label_3  label_4
0       45      NaN        1        6
1       45      NaN        1        6
2       45      NaN        1        6
3       45      NaN        1        6
4       45      NaN        1        6

# KNN Model


In [24]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import GridSearchCV
import numpy as np
from sklearn.metrics import accuracy_score

def try_knn(train_x, train_y, valid_x, valid_y):
    grid_params = { 'n_neighbors' : [3,5,7,15],
               'weights' : ['uniform','distance']}
    gs = GridSearchCV(KNeighborsClassifier(), grid_params, verbose = 1, cv=3, n_jobs = -1)
    g_res = gs.fit(train_x, train_y)
    print(f"Best Hyperparameters:",gs.best_params_)
    # Get the best k-NN model with the optimal hyperparameters
    best_knn = gs.best_estimator_
    # Evaluate the best model on the test data
    accuracy = best_knn.score(valid_x, valid_y)
    print(f"Accuracy for KNN {accuracy}")

# SVM Model

In [17]:
from sklearn import svm
from sklearn.metrics import accuracy_score

def try_svm(train_x, train_y, valid_x, valid_y):
    clf = svm.SVC()
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with default settings (RBF - exponential kernal) ", accuracy_score(valid_y, y_pred))
    
    clf = svm.LinearSVC(dual="auto")
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of Linear SVM (one-vs-the-rest) ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='linear')
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with linear kernal function ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='sigmoid')
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with sigmoid kernal function ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='poly',degree =2)
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with polynomial kernal function with degree 2 ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='poly',degree =2)
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with polynomial kernal function with degree 3 ", accuracy_score(valid_y, y_pred))

# xgBoost Model

In [78]:
import xgboost as xgb
from sklearn.preprocessing import LabelEncoder

def try_xgBoost(train_x, train_y, vaid_x, valid_y):
    # Create an XGBoost classifier for multi-class classification
    clf = xgb.XGBClassifier(
    objective='multi:softmax',  # Set the objective for multi-class classification
    num_class=len(np.unique(train_y)),  # Number of classes
    random_state=42
    )
    le = LabelEncoder()
    train_y = le.fit_transform(train_y)
    clf.fit(train_x, train_y)
    y_pred = clf.predict(vaid_x)
    y_pred = le.inverse_transform(y_pred)
    print(f"Accuracy Score of XG Boost", accuracy_score(valid_y, y_pred))

# Rondom Forest Model

In [71]:
from sklearn.ensemble import RandomForestClassifier

def try_random_forest(train_x, train_y, vaid_x, valid_y):
    clf = RandomForestClassifier(n_estimators=100, random_state=42)
    clf.fit(train_x, train_y)
    y_pred = clf.predict(vaid_x)
    print(f"Accuracy Score of Random Forest", accuracy_score(valid_y, y_pred))

# Logistic Regression Model

In [72]:
from sklearn.linear_model import LogisticRegression

def try_logistic(train_x, train_y, vaid_x, valid_y):
    for i in ["lbfgs","newton-cg","sag","saga"]:
        logistic_regression = LogisticRegression(multi_class='multinomial', solver='lbfgs')
        logistic_regression.fit(train_x,train_y)
        y_pred = logistic_regression.predict(vaid_x)
        print(f"Accuracy Score of Logistic Regression with solver {i}", accuracy_score(valid_y, y_pred))

# Predict a probability for the TARGET variable

In [28]:
test_df=pd.read_csv('test.csv')
test_df.head()

ID  feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   1   0.187868  -0.009268   0.039846  -0.102515  -0.109121  -0.079769   
1   2   0.122119   0.019015   0.134483   0.007949  -0.157884  -0.033332   
2   3   0.079187   0.146195  -0.001316  -0.085596  -0.261093  -0.021620   
3   4   0.231196   0.021558   0.215534  -0.210258  -0.158189   0.041621   
4   5   0.020412   0.063732  -0.065506  -0.089598  -0.130788  -0.018809   

   feature_7  feature_8  feature_9  ...  feature_759  feature_760  \
0   0.061295  -0.042890   0.099860  ...     0.043647     0.061174   
1   0.044334  -0.016869  -0.014088  ...    -0.179798    -0.027611   
2   0.268898  -0.103950   0.068976  ...    -0.075460     0.037346   
3   0.240254  -0.068112  -0.017550  ...     0.083334     0.141642   
4   0.119304  -0.057494   0.094714  ...    -0.001969    -0.003025   

   feature_761  feature_762  feature_763  feature_764  feature_765  \
0     0.026771    -0.092734    -0.052383    -0.055526     0.029101   
1     0.240023    -0.005648    -0.057537     0.023922    -0.038126   
2     0.124254    -0.369695    -0.056773    -0.028471     0.019971   
3    -0.007219     0.031547    -0.010762    -0.169036    -0.030963   
4    -0.021156    -0.136993    -0.003615     0.036371     0.048715   

   feature_766  feature_767  feature_768  
0     0.041857    -0.027872     0.099500  
1    -0.015171    -0.006270    -0.003790  
2    -0.065683    -0.047366    -0.121744  
3     0.086698     0.114194    -0.036775  
4    -0.002688    -0.016957    -0.004331  

[5 rows x 769 columns]

In [33]:
X_test = test_df.drop(["ID"], axis=1)

In [34]:
X_test.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.187868  -0.009268   0.039846  -0.102515  -0.109121  -0.079769   
1   0.122119   0.019015   0.134483   0.007949  -0.157884  -0.033332   
2   0.079187   0.146195  -0.001316  -0.085596  -0.261093  -0.021620   
3   0.231196   0.021558   0.215534  -0.210258  -0.158189   0.041621   
4   0.020412   0.063732  -0.065506  -0.089598  -0.130788  -0.018809   

   feature_7  feature_8  feature_9  feature_10  ...  feature_759  feature_760  \
0   0.061295  -0.042890   0.099860   -0.066319  ...     0.043647     0.061174   
1   0.044334  -0.016869  -0.014088   -0.060542  ...    -0.179798    -0.027611   
2   0.268898  -0.103950   0.068976   -0.062388  ...    -0.075460     0.037346   
3   0.240254  -0.068112  -0.017550   -0.233767  ...     0.083334     0.141642   
4   0.119304  -0.057494   0.094714   -0.047923  ...    -0.001969    -0.003025   

   feature_761  feature_762  feature_763  feature_764  feature_765  \
0     0.026771    -0.092734    -0.052383    -0.055526     0.029101   
1     0.240023    -0.005648    -0.057537     0.023922    -0.038126   
2     0.124254    -0.369695    -0.056773    -0.028471     0.019971   
3    -0.007219     0.031547    -0.010762    -0.169036    -0.030963   
4    -0.021156    -0.136993    -0.003615     0.036371     0.048715   

   feature_766  feature_767  feature_768  
0     0.041857    -0.027872     0.099500  
1    -0.015171    -0.006270    -0.003790  
2    -0.065683    -0.047366    -0.121744  
3     0.086698     0.114194    -0.036775  
4    -0.002688    -0.016957    -0.004331  

[5 rows x 768 columns]

In [37]:
X_test.isnull().sum()

feature_1      0
feature_2      0
feature_3      0
feature_4      0
feature_5      0
              ..
feature_764    0
feature_765    0
feature_766    0
feature_767    0
feature_768    0
Length: 768, dtype: int64

# Label_1


# Use the SVM Model for label_1

In [18]:
try_svm(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.956


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.9826666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.9746666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.8973333333333333


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.956


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 3  0.956


# Use the KNN Model for label_1

In [26]:
try_knn(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 3, 'weights': 'distance'}
Accuracy for KNN 0.948


In [79]:
try_xgBoost(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

Accuracy Score of XG Boost 0.9026666666666666


In [74]:
try_random_forest(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

Accuracy Score of Random Forest 0.9066666666666666


In [75]:
try_logistic(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.972


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.972


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.972
Accuracy Score of Logistic Regression with solver saga 0.972


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


# For label_1 we get best accuracy(0.98266) when using the Linear SVM
So we can use this method to predict the values for test data set.

In [35]:
clf = svm.SVC(kernel='linear')
clf.fit(X_train, y_train["label_1"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label1 = clf.predict(X_test_contiguous)
y_pred_label1

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


array([26, 18, 16,  7, 58, 46,  7, 22, 29, 26, 33,  6, 54, 51, 29, 48, 23,
        2, 43, 42, 11, 35, 39, 36, 44, 48, 28, 55,  2, 51, 55, 46,  1, 56,
       30, 22, 50,  6, 40, 18, 19, 47,  6, 46, 59, 25, 31, 38,  5, 32, 25,
       56, 39, 37, 32, 29, 46, 52, 28, 34, 16, 47, 55, 24, 23, 18, 20,  6,
       36, 38, 22, 28, 44, 46, 55, 59, 49, 55, 57, 12, 52, 38, 29,  4, 31,
       49, 14, 19, 12, 45, 36, 43, 34, 32, 54, 46, 27,  5,  7, 42, 19, 26,
       31, 36, 53, 10, 12, 33, 38,  2, 57, 37, 17, 50, 46, 46, 17, 34, 32,
       40, 19, 40,  8, 22, 38, 15, 39,  9, 52, 37, 28, 13, 10, 55, 28, 25,
       42, 57, 16, 27, 23,  3, 51, 27, 32, 46, 16, 53, 11, 46, 42, 37, 47,
       50,  3, 41, 19, 20, 15,  8,  3, 18,  5, 28, 46, 49, 16, 40, 17, 20,
       53, 57,  7, 16, 11, 35, 34, 52, 41, 55,  7, 18, 21, 19, 37, 55, 29,
       28, 23, 35, 46, 52, 30, 27, 13,  7, 28, 14,  3, 36, 17, 60, 18,  3,
       22,  9, 46, 25, 17, 26, 19, 36, 36, 21,  1, 44, 39, 47, 38,  6, 50,
       39, 23, 37, 48, 22

# Label_2

In [49]:
df_label_2 = df.dropna(subset=['label_2'])
df_label_2.isnull().sum()

feature_1      0
feature_2      0
feature_3      0
feature_4      0
feature_5      0
              ..
feature_768    0
label_1        0
label_2        0
label_3        0
label_4        0
Length: 772, dtype: int64

In [51]:
y_train_l2 = df_label_2[["label_1","label_2","label_3","label_4"]]
X_train_l2 = df_label_2.drop(y_train_l2, axis=1)

In [53]:
y_train_l2.head()

label_1  label_2  label_3  label_4
480        5     25.0        1        6
481        5     25.0        1        6
482        5     25.0        1        6
483        5     25.0        1        6
484        5     25.0        1        6

In [61]:
valid_df_label_2 = valid_df.dropna(subset=['label_2'])
valid_df_label_2.isnull().sum()

feature_1      0
feature_2      0
feature_3      0
feature_4      0
feature_5      0
              ..
feature_768    0
label_1        0
label_2        0
label_3        0
label_4        0
Length: 772, dtype: int64

In [62]:
y_valid_l2 = valid_df_label_2[["label_1","label_2","label_3","label_4"]]
X_valid_l2 = valid_df_label_2.drop(y_train_l2, axis=1)

In [63]:
y_valid_l2.head()

label_1  label_2  label_3  label_4
14        5     25.0        1        6
15        5     25.0        1        6
16        5     25.0        1        6
17        5     25.0        1        6
18        5     25.0        1        6

# Use the SVM Model for label_2

In [64]:
try_svm(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.8804347826086957


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.8301630434782609


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.8546195652173914


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.485054347826087


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.873641304347826


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 3  0.873641304347826


# Use the KNN Model for label_2

In [66]:
try_knn(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 3, 'weights': 'distance'}
Accuracy for KNN 0.9470108695652174


In [84]:
try_xgBoost(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

Accuracy Score of XG Boost 0.8872282608695652


In [85]:
try_random_forest(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

Accuracy Score of Random Forest 0.782608695652174


In [86]:
try_logistic(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.7540760869565217


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.7540760869565217


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.7540760869565217
Accuracy Score of Logistic Regression with solver saga 0.7540760869565217


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


For label_4 we get best accuracy(0.94701) when using the KNN model. 
So we can use this methods to predict the values for test data set. 

In [68]:
k = 3
knn_model = KNeighborsClassifier(n_neighbors=k)
knn_model.fit(X_train_l2, y_train_l2["label_2"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label2 = knn_model.predict(X_test_contiguous)
y_pred_label2

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but KNeighborsClassifier was fitted with feature names
  warnings.warn(


array([22., 25., 30., 27., 29., 30., 27., 33., 23., 22., 26., 25., 27.,
       26., 23., 26., 28., 25., 22., 29., 33., 24., 29., 22., 61., 26.,
       28., 23., 25., 26., 23., 30., 30., 24., 28., 24., 24., 25., 26.,
       25., 26., 23., 25., 30., 31., 22., 26., 32., 25., 23., 22., 24.,
       29., 27., 23., 23., 30., 34., 28., 25., 30., 23., 23., 26., 28.,
       25., 25., 25., 22., 32., 33., 28., 61., 30., 23., 31., 26., 26.,
       27., 26., 34., 32., 23., 23., 26., 26., 31., 23., 26., 29., 29.,
       31., 25., 23., 27., 30., 31., 25., 27., 29., 23., 22., 26., 22.,
       24., 36., 26., 26., 32., 25., 27., 27., 26., 24., 30., 30., 26.,
       25., 25., 26., 26., 26., 41., 33., 32., 28., 29., 35., 34., 25.,
       28., 27., 36., 23., 28., 22., 29., 27., 30., 31., 23., 25., 26.,
       31., 23., 30., 30., 23., 33., 30., 29., 27., 23., 24., 31., 30.,
       23., 25., 28., 23., 31., 25., 25., 28., 30., 25., 30., 26., 26.,
       25., 23., 27., 31., 30., 33., 24., 25., 34., 28., 23., 27

# Label_3


# Use the SVM Model for label_3

In [36]:
try_svm(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.9986666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.9986666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.9986666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.9346666666666666


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.9986666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 3  0.9986666666666667


# Use the KNN Model for label_3

In [38]:
try_knn(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 7, 'weights': 'distance'}
Accuracy for KNN 0.9973333333333333


In [80]:
try_xgBoost(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

Accuracy Score of XG Boost 0.996


In [76]:
try_random_forest(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

Accuracy Score of Random Forest 0.9613333333333334


In [82]:
try_logistic(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.9986666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.9986666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.9986666666666667
Accuracy Score of Logistic Regression with solver saga 0.9986666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


For label_3 we get best accuracy(0.9986) when using the SVM with default settings, Linear SVM, SVM with linear kernal, SVM with polynomial kernal and all logistic models.
So we can use any of these methods to predict the values for test data set. I am going to use Linear SVM.

In [39]:
clf = svm.SVC(kernel='linear')
clf.fit(X_train, y_train["label_3"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label3 = clf.predict(X_test_contiguous)
y_pred_label3

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


array([0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1,
       1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1,
       1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1,
       0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1,
       1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1,
       1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1,
       1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 1, 1,
       1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0,
       1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
       1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,

# Label_4

# Use the SVM Model for label_4

In [41]:
try_svm(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.9426666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.9413333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.9573333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.7653333333333333


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.9413333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 3  0.9413333333333334


# Use the KNN Model for label_4

In [42]:
try_knn(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 7, 'weights': 'distance'}
Accuracy for KNN 0.9626666666666667


In [81]:
try_xgBoost(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

Accuracy Score of XG Boost 0.92


In [77]:
try_random_forest(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

Accuracy Score of Random Forest 0.7666666666666667


In [83]:
try_logistic(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.9306666666666666


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.9306666666666666


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.9306666666666666
Accuracy Score of Logistic Regression with solver saga 0.9306666666666666


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


For label_4 we get best accuracy(0.96266) when using the KNN model
So we can use this methods to predict the values for test data set. 

In [46]:
k = 7
knn_model = KNeighborsClassifier(n_neighbors=k)
knn_model.fit(X_train, y_train["label_4"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label4 = knn_model.predict(X_test_contiguous)
y_pred_label4

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but KNeighborsClassifier was fitted with feature names
  warnings.warn(


array([ 2,  8,  6,  6,  6,  6,  6,  6,  6,  2,  6,  6,  6,  6,  6,  6,  6,
        6,  5,  0,  6,  2,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6,  8,  6,  3,  6,  6,  6,  1,  6, 12,  6,  0,  1,
        6,  6,  7,  0,  6,  6,  5,  6,  6,  6,  3,  6,  2,  6,  6,  6,  6,
        6, 12,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  5, 12,  6,  6,  6,
        6, 12,  4,  6,  6,  6,  6,  6,  0,  6,  6,  6,  6,  6,  0,  4,  2,
        6,  6,  6,  6,  6,  6, 12,  6,  6,  7,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6, 12,  9,  6, 11,  5,  6,  6,  6,  6,  6,  6,  1,
        0,  6,  6,  7,  6,  6,  6,  7,  0,  6,  6,  6,  6,  6,  0,  7,  3,
        6,  6, 10,  4,  6,  9,  6,  6,  8,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  7,  6,  6,  6,  6,  5,  9,  6,  6,  8,  6,  4,  7,  6,  6,
        6,  6,  2,  6,  5,  6,  7,  6,  6,  6, 12,  6,  6,  6, 13,  8,  6,
        6, 11,  6,  1,  6,  2,  6,  6,  6,  6,  6,  6,  6,  3, 12,  6,  6,
        6,  6,  7,  6,  6

# Final output

In [69]:
final_df = pd.concat([pd.Series(test_df['ID']), pd.Series(y_pred_label1, name='label_1'), pd.Series(y_pred_label2, name='label_2'), pd.Series(y_pred_label3, name='label_3'), pd.Series(y_pred_label4, name='label_4')], axis=1)
output_file_name = "190541R_lab_2.csv"
final_df.to_csv(output_file_name, index=False)
